src/server/dflycmd.cc

```diff
@@ -326,49 +311,56 @@ void DflyCmd::Expire(CmdArgList args, ConnectionContext* cntx) {
   return rb->SendOk();
 }
 
-OpStatus DflyCmd::StartFullSyncInThread(FlowInfo* flow, EngineShard* shard) {
-  DCHECK(!flow->fb.joinable());
+OpStatus DflyCmd::StartFullSyncInThread(FlowInfo* flow, Context* cntx, EngineShard* shard) {
+  DCHECK(!flow->task_fb.joinable());
 
   SaveMode save_mode = shard == nullptr ? SaveMode::SUMMARY : SaveMode::SINGLE_SHARD;
   flow->saver.reset(new RdbSaver(flow->conn->socket(), save_mode, false));
 
+  flow->cleanup = [saver = flow->saver.get()]() { saver->Cancel(); };
+
   // Shard can be null for io thread.
   if (shard != nullptr) {
-    auto ec = sf_->journal()->OpenInThread(false, string_view());
-    CHECK(!ec);
-    flow->saver->StartSnapshotInShard(true, shard);
+    CHECK(!sf_->journal()->OpenInThread(false, ""sv));  // can only happen in persistent mode.
+    flow->saver->StartSnapshotInShard(true, *cntx, shard);
   }
 
-  flow->fb = ::boost::fibers::fiber(&DflyCmd::FullSyncFb, this, flow);
+  flow->task_fb = ::boost::fibers::fiber(&DflyCmd::FullSyncFb, this, flow, cntx);
```

Comment:
are you planning to assign it different functions?

Reply:
Right now, no. Should I give it a more distinct name connected to its only task? Like full_sync_fb?